### PR TITLE
 Add newline to news files created with create-news-file 

### PR DESCRIPTION
**Pull request by rwalton-arm**
### Description

Add newline to news files created with create-news-file 

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_tools_ci_scripts/create_news_file.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def determine_news_file_path(news_type: NewsType) -> pathlib.Path:
 
 def _write_file(file_path: pathlib.Path, text: str) -> None:
     file_path.parent.mkdir(parents=True, exist_ok=True)
+    if not text.endswith("\n"):
```
Comment (Patater) [inline mbed_tools_ci_scripts/create_news_file.py:58]: Why `if`? Can't we always add the `\n`? Don't the tests in the next commit always assume a `\n` is added?

**Review by urutva — APPROVED** [review]

File: mbed_tools_ci_scripts/create_news_file.py (hunk 1 above)
Comment (rwalton-arm) [reply mbed_tools_ci_scripts/create_news_file.py:58]: If we add two newlines then `splitlines` returns a list of `len() == 2` and the `assert-news` check will fail.

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/create_news_file.py:58]: https://github.com/ARMmbed/mbed-tools-ci-scripts/blob/master/mbed_tools_ci_scripts/assert_news.py#L47

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/create_news_file.py:58]: But we probably should test it doesn't add a newline if one already exists. I'll add another test case for that.

File: tests/test_create_news_file.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 import pathlib
-from pyfakefs.fake_filesystem_unittest import Patcher
 from unittest import TestCase, mock
 from datetime import datetime
 from tempfile import TemporaryDirectory
```
Comment (rwalton-arm) [inline tests/test_create_news_file.py:8]: Reduces test time from 1.58s to 0.37s.

File: mbed_tools_ci_scripts/create_news_file.py (hunk 1 above)
Comment (Patater) [reply mbed_tools_ci_scripts/create_news_file.py:58]: `If we add two newlines` but we wouldn't add two newlines. Is there a case where we would?

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/create_news_file.py:58]: If you give "my newsfile message\n" to it on the command line :)

Comment (Patater) [reply mbed_tools_ci_scripts/create_news_file.py:58]: bah, who does that :p

**Review by Patater — APPROVED** [review]

**Pull request merged**